### PR TITLE
Merge unified/dev into dev

### PR DESCRIPTION
**Pull request by heidijinxujia**

### CODE REVIEW

**Comment by heidijinxujia** [comment]
All unit tests and build passed on my local machine.

**Comment by heidijinxujia** [comment]
TravisCI reports compileReleaseJavaWithJavac errors. However, I checked the code and run on my local machine. All green.

**Comment by iambmelt** [comment]
Changes from https://github.com/AzureAD/microsoft-authentication-library-for-android/commit/faf8357a2e18471c445b82847a0c18b24faf30c5 do not appear to have merged properly (constructor changes missing).

There may be other work which was 'lost' in this merge, but this is the first instance I found

File: msal/build.gradle [hunk 1]
```diff
@@ -106,59 +103,21 @@ task sourcesJar(type: Jar) {
...
-    androidTestCompile "com.android.support.test:runner:$rootProject.ext.runnerVersion"
+    androidTestImplementation "com.android.support.test:runner:$rootProject.ext.runnerVersion"
     // Set this dependency to use JUnit 4 rules
-    androidTestCompile "com.android.support.test:rules:$rootProject.ext.rulesVersion"
-    androidTestCompile "org.mockito:mockito-core:$rootProject.ext.mockitoCoreVersion"
-    androidTestCompile "com.google.dexmaker:dexmaker:$rootProject.ext.dexmakerMockitoVersion"
-    androidTestCompile "com.google.dexmaker:dexmaker-mockito:$rootProject.ext.dexmakerMockitoVersion"
-}
-
-task createPom {
```
[the remark this answers is not in the document]
Comment (iambmelt) [reply msal/build.gradle]: Issue created here: https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/311

File: msal/build.gradle [hunk 2]
```diff
@@ -63,7 +60,7 @@ task jacocoTestReport(type: JacocoReport, dependsOn: 'testDebugUnitTest') {
         html.enabled = true
     }
 
-    jacocoClasspath = configurations['androidJacocoAnt']
+    jacocoClasspath = configurations['jacocoAnt']
```
[the remark this answers is not in the document]
Comment (iambmelt) [reply msal/build.gradle]: https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/312

**Pull request merged**